### ISSUE TITLE
Build failure after update from 1.1.0 to 1.3.0 (or 1.2.0): Type androidx.appcompat.app.ActionBar$DisplayOptions is defined multiple times

### ISSUE THREAD

**Issue by opatry** [material-components/material-components-android#2166]
**Description:** When building in release both app & instrumented tests with material 1.2.0+, build fails.
```
Execution failed for task ':app:mergeExtDexRelease'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
     Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
     Type androidx.appcompat.app.ActionBar$DisplayOptions is defined multiple times: /home/build/.gradle/caches/transforms-3/230d761abc2a9fbf4dacd2f7dafa174e/transformed/appcompat-1.2.0-runtime/classes.dex, /home/build/.gradle/caches/transforms-3/fa06e5dc648e3583f0d86da56f60831d/transformed/appcompat-1.2.0-runtime/classes.dex
```

**Expected behavior:** BUILD SUCCESSFUL

**Source code:** Sample project to reproduce the issue available here: https://github.com/opatry/desugaring-issue

**TL;DR**

`app/build.gradle`
```
android {
...

    testBuildType System.getProperty('buildType') ?: 'debug'

    compileOptions {
        // WORKAROUND: enable coreLibraryDesugaringEnabled
//        coreLibraryDesugaringEnabled true
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }
    kotlinOptions {
        jvmTarget = '1.8'
    }
}

dependencies {
    // WORKAROUND: enable coreLibraryDesugaringEnabled
//    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.5'

    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${kotlin_version}"
    implementation 'com.google.android.material:material:1.3.0'
    implementation 'androidx.preference:preference:1.1.1'
    androidTestImplementation 'androidx.test.espresso:espresso-contrib:3.3.0'
}
```

`gradle.properties`
```
android.enableJetifier=false
```

Then
```
./gradlew assembleRelease app:assembleReleaseAndroidTest -DbuildType=release
```

**Android API version:** 30

**Material Library version:** 1.3.0 (or 1.2.0, fine with 1.1.0)

**Device:** n/a (build time issue)

**Comment by opatry** [comment]
Given this analysis, I finally succeeded finding an existing bug on the AGP tracker https://issuetracker.google.com/issues/173043489
This bug is closed because related to a Gradle bug https://github.com/gradle/gradle/issues/15536, closed itself.

It appears it should be fixed with Gradle 7.
In the meantime, @alaershov suggested the following workaround:
```
tasks.named('mergeExtDexDebugAndroidTest').configure {
    mustRunAfter tasks.named('mergeExtDexDebug')
}
```
But this doesn't work on my setup.

**Comment by drchen** [comment]
Close the issue due to it's not a Material library issue.
